# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../gentoo/gtk.patch
# Copyright (C) 2010 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- gentoo-0.15.3.orig/src/odmultibutton.c
+++ gentoo-0.15.3/src/odmultibutton.c
@@ -73,8 +73,8 @@
 		}
 		if(p != NULL)
 		{
-			if(GTK_WIDGET_STATE(widget) != GTK_WIDGET_STATE(p))
-				gtk_widget_set_state(p, GTK_WIDGET_STATE(widget));
+			if(gtk_widget_get_state(widget) != gtk_widget_get_state(p))
+				gtk_widget_set_state(p, gtk_widget_get_state(widget));
 			gtk_widget_show(p);
 			if(gtk_widget_get_parent(p) != NULL)
 				gtk_widget_reparent(p, widget);
@@ -86,7 +86,7 @@
 		}
 		mb->last_index = index;
 	}
-	if(GTK_WIDGET_DRAWABLE(widget))
+	if(gtk_widget_is_drawable(widget))
 		gtk_widget_queue_draw(widget);
 }
 
@@ -144,9 +144,9 @@
 {
 	GtkStateType	state;
 
-	state = GTK_WIDGET_STATE(widget);
+	state = gtk_widget_get_state(widget);
 
-	if(GTK_WIDGET_DRAWABLE(widget) && state != GTK_STATE_ACTIVE)
+	if(gtk_widget_is_drawable(widget) && state != GTK_STATE_ACTIVE)
 	{
 		GtkStyle	*style = gtk_widget_get_style(widget);
 		const guint	EARSIZE = 5;
@@ -209,7 +209,7 @@
 	g_return_if_fail(area != NULL);
 	g_return_if_fail(OD_IS_MULTIBUTTON(widget));
 
-	if(GTK_WIDGET_DRAWABLE(widget))
+	if(gtk_widget_is_drawable(widget))
 	{
 		ODMultiButton	*mb = OD_MULTIBUTTON(widget);
 
@@ -449,7 +449,7 @@
 			od_multibutton_reset_label(mb, index, GTK_LABEL(w), text, bg, fg);
 			if(GTK_WIDGET(mb)->parent)
 				gtk_widget_queue_resize(GTK_WIDGET(mb));
-			if(GTK_WIDGET_DRAWABLE(mb))
+			if(gtk_widget_is_drawable(mb))
 				gtk_widget_queue_draw(GTK_WIDGET(mb));
 		}
 	}

